alica_engine/include/engine/AlicaContext.h

```diff
@@ -417,6 +424,17 @@ class AlicaContext
      * @note Is called when setOption or setOptions is successfully called.
      */
     void reloadConfig();
+
+    /*
+    Communication Handlers
+    */
+    void handleOnSyncTalk(std::shared_ptr<SyncTalk> st);
+    void handleOnSyncReady(std::shared_ptr<SyncReady> sr);
+    void handleIncomingAuthorityMessage(const AllocationAuthorityInfo& aai);
+    void handlePlanTreeInfo(std::shared_ptr<PlanTreeInfo> st);
+    void handleOnSolverResult(const SolverResult& sr);
+    void handleAgentQuery(const AgentQuery& pq);
+    void handleAgentAnnouncement(const AgentAnnouncement& pa);
```

Comment:
you don't need these private methods. Just pass lambdas that forward calls to alica engine while creating AlicaCommunicationCallbacks struct in setCommunicator method

Reply:
using lambda's would imply that we need to access the engine internals & so can't forward declare the engine anymore. Right now these callbacks are implemented in the .cpp file so that the header can just forward declare the engine as it was earlier (which was probably done to save compilation time)